cli/src/commands/git.rs

```diff
@@ -742,38 +742,28 @@ fn cmd_git_push(
     let mut tx = workspace_command.start_transaction();
     let tx_description;
     let mut branch_updates = vec![];
-    if args.all {
-        for (branch_name, targets) in repo.view().local_remote_branches(&remote) {
+    let populate_branch_updates_skip_missing =
+        |(branch_name, targets)| -> Result<(), CommandError> {
             match classify_branch_update(branch_name, &remote, targets) {
                 Ok(Some(update)) => branch_updates.push((branch_name.to_owned(), update)),
                 Ok(None) => {}
                 Err(reason) => reason.print(ui)?,
-            }
-        }
+            };
+            Ok(())
+        };
+    let mut local_remote_branches = repo.view().local_remote_branches(&remote);
```

Comment:
nit: I would inline `repo.view().local_remote_branches(&remote)` or make it a closure so the consumed iterator will never be reused by mistake.

Reply:
Sure, I can inline it back. I found there is one use of it that I can't switch to using this variable anyway.